x-pack/plugins/enterprise_search/server/utils/create_pipeline_definitions.test.ts

```diff
@@ -34,3 +35,163 @@ describe('createIndexPipelineDefinitions util function', () => {
     expect(mockClient.ingest.putPipeline).toHaveBeenCalledTimes(3);
   });
 });
+
+describe('formatMlPipelineBody util function', () => {
+  const modelId = 'my-model-id';
+  let modelInputField = 'my-model-input-field';
+  const modelType = 'my-model-type';
+  const modelVersion = 3;
+  const sourceField = 'my-source-field';
+  const destField = 'my-dest-field';
+
+  const mockClient = {
+    ml: {
+      getTrainedModels: jest.fn(),
+    },
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should return the pipeline body', async () => {
+    const expectedResult = {
+      description: '',
+      version: 1,
+      processors: [
+        {
+          remove: {
+            field: `ml.inference.${destField}`,
+            ignore_missing: true,
+          },
+        },
+        {
+          inference: {
+            model_id: modelId,
+            target_field: `ml.inference.${destField}`,
+            field_map: {
+              sourceField: modelInputField,
```

Comment:
@j-bennet Do we want to pass `"sourceField": <model input field>`, literally with this key? Or the _value_ of `sourceField` from the args? If the latter, we need to use this notation:
```
[sourceField]: modelInputField,
```

I can fix this in my upcoming PR if there's a need.

Reply:
The value of `sourceField`.

Reply:
What are the square brackets for?

Reply:
If `sourceField` is a variable, then the square brackets will make the key the value of the `sourceField`:
```node
> const sourceField = 'myVariable';
undefined

> { sourceField: 'value' }
{ sourceField: 'value' }

> { [sourceField]: 'value' }
{ myVariable: 'value' }
```